cmd/crossplane/core/core.go

```diff
@@ -209,6 +209,12 @@ func (c *startCommand) Run(s *runtime.Scheme, log logging.Logger) error { //noli
 		log.Info("Extra Resources are GA and cannot be disabled. The --enable-composition-functions-extra-resources flag will be removed in a future release.")
 	}
 
+	if c.EnableEnvironmentConfigs {
+		// TODO(negz): Include a link to a migration guide.
+		// TODO(negz): Perhaps this should return an error instead of logging?
+		log.Info("Crossplane no longer supports loading and patching EnvironmentConfigs natively. Please use function-environment-configs instead. The --enable-environment-configs flag will be removed in a future release.")
```

Comment:
I suspect it would be best for this to return an error, and have the error link to a migration guide.

That way folks who don't notice that we've dropped support and try to upgrade will get a "clean" error rather than having Crossplane potentially enter a half-working state.

Reply:
They will still get an error as soon as they start trying to apply compositions using dropped fields.

Reply:
Yeah. I was thinking more about existing Compositions though. I think the environment related code would just stop working, right?

Reply:
Yes, true, can't think of a nice workaround though. The only option would be to add a pre-upgrade hook to the chart or better a in the init container installing the CRDs, erroring out if a composition is using environment related functionalities.